feat: new provider postbin

GangGreenTemperTatum opened the pull request:
superseeds #10 
adds another public provider [https://pastebin.com/](https://pastebin.com/) 🥳 

usage:

https://github.com/user-attachments/assets/08811732-300f-4c1f-abb3-20cd04ae839b

thanks again @hahwul 🚀

hahwul commented:
@GangGreenTemperTatum 
Thanks for the PR! I have two issues to address and one suggestion:

## Auto-Generated Payload (Postbin, Webhook.site)

The README suggests that Webhook.site and Postbin support auto-generated payloads. However, after clicking `Add Public Postbin Provider` and selecting "Get Payload" in the Oast tab, no payload is returned. The same happens with Webhook.site. Is this expected, or might something be missing?

## Typecheck Issue

There's a type error in `postbin.ts`:

```
pnpm typecheck
# ...
# packages/backend typecheck$ tsc --noEmit
# src/services/postbin.ts(94,23): error TS2345: Argument of type '{ id: any; type: string; destination: string | null; timestamp: Date; ... }' ...
#   Types of property 'timestamp' are incompatible: 'Date' is not assignable to 'number'.
```

Looks like the timestamp type needs adjustment. Could you take a look?

## Quick Button Simplification

Quick Buttons for BOAST and Interactsh are handy for testing. As more providers are added, though, it might be worth simplifying this area—perhaps switching to icons or a dropdown? Also, adding an "auto-generate" button directly to "Add Provider" could streamline the UX. What do you think?

GangGreenTemperTatum commented:
> There's a type error in `postbin.ts`:
> 
> ```
> pnpm typecheck
> # ...
> # packages/backend typecheck$ tsc --noEmit
> # src/services/postbin.ts(94,23): error TS2345: Argument of type '{ id: any; type: string; destination: string | null; timestamp: Date; ... }' ...
> #   Types of property 'timestamp' are incompatible: 'Date' is not assignable to 'number'.
> ```

thanks so much for the feedback as always @hahwul !

1. Addressed in [fix: get payload om webhook and postbin and updated polling functions](https://github.com/hahwul/OmniOAST/pull/11/commits/948bf77a087a7c0d992038b420c3d9ee80522912) - tested and confirmed working ✅ 
2. Typescript error also addressed in the same commit. The typecheck is now passing successfully. ✅ 
 - This ensures the timestamp is always a number, which matches the OASTEvent interface requirement. 
```typescript
timestamp: request.inserted ? new Date(request.inserted).getTime() : Date.now(),
```

3. love your suggestion and to push further here, what do you think to this new UX attempt? ([feat: enhance ux with dropdown quick add provider](https://github.com/hahwul/OmniOAST/pull/11/commits/1d9c6d06a282171130791d3e8d59726f49a209ae))

<img width="2435" height="950" alt="image" src="https://github.com/user-attachments/assets/70088d08-bf0c-481d-b7b6-e0aeeff22bfc" />

also - no pressure here and out of curiosity - would you be happy expanding the [authors](https://github.com/GangGreenTemperTatum/OmniOAST/blob/948bf77a087a7c0d992038b420c3d9ee80522912/caido.config.ts#L17-L22) that i could include myself for the next release? if so, i can add on this PR too

```typescript
  author: {
    name: "hahwul",
    email: "hahwul@gmail.com",
    url: "https://www.hahwul.com",
  },
```

hahwul commented:
@GangGreenTemperTatum 
Looks great! It works well and the features and code are awesome. I'll merge it!

Regarding your question, it seems that the current [Caido Plugin structure](https://github.com/caido-community/dev/blob/5d4b692751d2b83211d0c7aaaea187d519ca2b56/src/manifest.ts#L12) only allows a single user to be added as the author. I'll note it as a TODO and discuss with the Caido team to see if we can extend it to support an array or include main contributor mentions!

GangGreenTemperTatum commented:
> @GangGreenTemperTatum Looks great! It works well and the features and code are awesome. I'll merge it!
> 
> Regarding your question, it seems that the current [Caido Plugin structure](https://github.com/caido-community/dev/blob/5d4b692751d2b83211d0c7aaaea187d519ca2b56/src/manifest.ts#L12) only allows a single user to be added as the author. I'll note it as a TODO and discuss with the Caido team to see if we can extend it to support an array or include main contributor mentions!

sweet thanks brother, appreciate it! 🥳 wooot!!

ahh gotcha, this makes sense.. thanks so much i really appreciate that and would love to continue future collabs 🚀 🤝
